File: backend/app/schemas/resume.py
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel
from app.db.enums import ResumeStatus

class ResumeResponse(BaseModel):
    id: UUID
    user_id: UUID
    original_filename: str
    stored_filename: str
    storage_path: str
    file_size: int
    mime_type: str
    status: ResumeStatus
    parsed_at: datetime | None = None
    processed_at: datetime | None = None
    parser_version: str | None = None
    uploaded_at: datetime
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class ResumeListResponse(BaseModel):
    resumes: list[ResumeResponse]

class ResumeUploadResponse(BaseModel):
    message: str
    resume: ResumeResponse

class ResumeUpdateResponse(BaseModel):
    message: str
    resume: ResumeResponse

File: frontend/src/types/resume.ts
export type ResumeStatus = "pending" | "processing" | "completed" | "failed";

export interface Resume {
  id: string;
  user_id: string;
  original_filename: string;
  stored_filename: string;
  storage_path: string;
  file_size: number;
  mime_type: string;
  status: ResumeStatus;
  parsed_at: string | null;
  processed_at: string | null;
  parser_version: string | null;
  uploaded_at: string;
  created_at: string;
  updated_at: string;
}

export interface ResumeListResponse {
  resumes: Resume[];
}

export interface ResumeUploadResponse {
  message: string;
  resume: Resume;
}

export interface ResumeUpdateResponse {
  message: string;
  resume: Resume;
}
